### PR TITLE
fix: error handling and error propagation back to the caller

### PR DESCRIPTION
**Pull request by MohdRafiUddin**
* fix: error handling and error propagation back to the caller (#13) [ENG-3136]

[ENG-3136]: https://cloudelements.atlassian.net/browse/ENG-3136

### CODE REVIEW

File: test/formulas/importFormulas.test.js [hunk 1]
```diff
@@ -24,8 +24,35 @@ describe('importFormulas', () => {
       processId: 2,
     });
   });
+  it('should be able to handle for empty file or directory', async () => {
+    let formulasData = await readFormulasFromDir(formulasDirectoryPath);
+    expect(formulasData).not.toBeNull();
+    formulasData = mapIndex(
```
Comment (mukundbayyaram) [inline test/formulas/importFormulas.test.js:30]: `formulasData` is not being used anywhere. Is this testing existence of some value or not needed?

File: src/actions/export.js [hunk 2]
```diff
@@ -30,11 +31,12 @@ module.exports = async (assetType, account, options) => {
     }
     eventListener.addListener();
     await exportByAssetType[assetType]({account, assetType, options});
+  } catch (error) {
+    logError(`An error occured during export operation for asset type: ${assetType}, error: ${error.message}`);
+    throw error;
+  } finally {
+    logDebug(`Successfully completed specified operation on core for asset type: ${assetType}`);
```
Comment (mukundbayyaram) [inline src/actions/export.js:38]: 🤞😅

File: src/core/elements/importElements.js [hunk 3]
```diff
@@ -68,31 +50,26 @@ const importElements = curry(async (elements, account, options) => {
...
+    } else if (pipe(prop('dir'), isNil, not)(options)) {
+      elementsDataToImport = await readElementsFromDir(options.dir);
+    }
+    if (isNilOrEmpty(elementsDataToImport)) {
+      logError(`No elements found for import operation`);
+      return;
+    }
+    if (pipe(type, equals('Object')) && pipe(prop('elements'), isNil, not)(elementsDataToImport)) {
+      /* istanbul ignore next */
+      await importElements(pipe(prop('elements'))(elementsDataToImport), account, options);
```
Comment (mukundbayyaram) [inline src/core/elements/importElements.js:67]: **nitpick:** We can do away without `pipe` here and in other operation files

**Review by mukundbayyaram — CHANGES_REQUESTED** [review]

File: test/formulas/importFormulas.test.js (hunk 1 above)
Comment (MohdRafiUddin) [reply test/formulas/importFormulas.test.js:30]: It's more like a flow that we perform on any operation/task and each test case represents some sort of different flows. You might see read data in some directory in each test case as it gets executed every time we call doctor-core.

File: src/core/elements/importElements.js (hunk 3 above)
Comment (MohdRafiUddin) [reply src/core/elements/importElements.js:67]: We can do that but I just wanted to keep this code as it's more readable for me.

**Review by mukundbayyaram — APPROVED** [review]

**Comment by mukundbayyaram** [comment]
Approved. I will merge it 
cc: @MohdRafiUddin

**Pull request merged**